drv/gimlet-seq-server/src/main.rs

```diff
@@ -1029,11 +1045,21 @@ impl<S: SpiServer> idl::InOrderSequencerImpl for ServerImpl<S> {
     }
 
     fn set_state(
+        &mut self,
```

Comment:
**suggestion:** Consider marking this as deprecated if the intention is that this should no longer be used.

Reply:
Hm, unfortunately, that's a bit tricky. We can't set an _implementation_ of a trait method as deprecated; instead, the trait itself must put the deprecated annotation on the definition of the trait method. Deprecating the method only in an `impl` block has no effect, and in fact, [results in a compiler error](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=20e207ebba3693a11c78832253974bc7). Because this is an Idol IPC interface, the trait `InOrderSequencerImpl` is not defined in Rust code anywhere, but is instead generated by Idol's codegen, so we can't easily add the deprecated attribute to it without changing the code generator to allow adding attributes to trait methods. And, *consumers* of this IPC interface don't actually interact with the `InOrderSequencerImpl` trait. Instead, that's used by the Idol server code to implement the IPC *server*. Clients of this interface interact with it through different generated code, so we would actually need to put the deprecated attribute on *that*.

It would be nice to have Idol support adding attributes such as `#[deprecated]` to generated client code, but that's something we'd have to go implement separately...